tests/testthat/_snaps/calculate.md

```diff
@@ -0,0 +1,137 @@
+
+    Code
+      calc_b <- calculate(b, values = draws)
+    Error <simpleError>
+      `nsim` must be set to sample <greta array>s not in MCMC samples
+      the target <greta array>s are related to new variables that are not in the MCMC
+      samples, so cannot be calculated from the samples alone.
+      Set `nsim` if you want to sample them conditionally on the MCMC samples
+
+# calculate errors nicely if non-greta arrays are passed
+
+    Code
+      calc_y <- calculate(y, x, values = list(x = c(2, 1)))
+    Error <simpleError>
+      `calculate()` arguments must be <greta_array>
+      The following object passed to `calculate()` is not a <greta array>:
+      "x"
+      Perhaps you forgot to explicitly name other arguments?
+      Perhaps you forgot to explicitly name other arguments?
+
+---
+
+    Code
+      calc_y <- calculate(y, list(x = c(2, 1)))
+    Error <simpleError>
+      `calculate()` arguments must be <greta_array>
+      The following object passed to `calculate()` is not a <greta array>:
+      "list(x = c(2, 1))"
+      Perhaps you forgot to explicitly name other arguments?
+      Perhaps you forgot to explicitly name other arguments?
+
+# calculate errors nicely if values for stochastics not passed
+
+    Code
+      calc_y <- calculate(y, values = list(x = c(2, 1)))
+    Error <simpleError>
+      Please provide values for the following 1 <greta_array>:
+      `a`
+
+# calculate errors nicely if values have incorrect dimensions
+
+    Code
+      calc_y <- calculate(y, values = list(a = c(1, 1)))
+    Error <simpleError>
+      a provided value has different number of elements than the <greta_array>
+
+# calculate errors nicely with invalid batch sizes
+
+    Code
+      calc_y <- calculate(y, values = draws, trace_batch_size = 0)
+    Error <simpleError>
+      `trace_batch_size` must be a single numeric value greater than or equal
+      to 1
+
+---
+
+    Code
+      calc_y <- calculate(y, values = draws, trace_batch_size = NULL)
+    Error <simpleError>
+      `trace_batch_size` must be a single numeric value greater than or equal
+      to 1
+
+---
+
+    Code
+      calc_y <- calculate(y, values = draws, trace_batch_size = NA)
+    Error <simpleError>
+      `trace_batch_size` must be a single numeric value greater than or equal
+      to 1
+
+# calculate errors if distribution-free variables are not fixed
+
+    Code
+      calc_a <- calculate(a, y, nsim = 1)
+    Error <simpleError>
+      the target <greta_array>s are related to variables that do not have
+      distributions so cannot be sampled
+
+# calculate errors if a distribution cannot be sampled from
+
+    Code
+      sims <- calculate(y, nsim = 1)
+    Error <simpleError>
+      sampling is not yet implemented for "hypergeometric" distributions
+
+# calculate errors nicely if nsim is invalid
+
+    Code
+      calc_x <- calculate(x, nsim = 0)
+    Error <simpleError>
+      value must be a positive integer
```

Comment:
Yup - that's because the error message is called from `check_positive_integer()`. I'm not sure how to get the argument out here, it might involve some NSE magic?

Reply:
I think that's what the `name` argument to `check_positive_integer()` used to do, you should be able to pass that through like this: `check_positive_integer(nsim, "nsim")` and make sure it's plumbed into the cli code. Though an NSE solution would be neat :)

Reply:
Ah that is neat! And avoids NSE magicks 🧙